python/ray/dataframe/dataframe.py

```diff
@@ -2882,35 +2881,72 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
                     are the quantiles.
         """
 
-        def quantile_helper(df, q, axis, numeric_only, interpolation):
-            try:
-                return df.quantile(q=q, axis=axis, numeric_only=numeric_only,
-                                   interpolation=interpolation)
-            except ValueError:
-                return pd.Series()
+        # deal with error checking for bad dtyping when numeric_only is False
+        if not numeric_only:
+            # check if there are any object columns
+            for t in self.dtypes:
+                if t == np.dtype('O'):
+                    e_msg = "Cannot call quantile on columns of type " + str(t)
+                    raise TypeError(e_msg)
+        else:
+            if all([dtype == np.dtype('O') or
+                    np.issubdtype(dtype, np.datetime64)
+                    for dtype in self.dtypes]):
+                raise ValueError("need at least one array to concatenate")
```

Comment:
I know that this error is concordant with Pandas, but it feels very out of place in this context. Perhaps we should change it to something more descriptive. Honestly, this case seems like a bug with Pandas, it should handle the case where all columns are dropped.

Looping in @devin-petersohn here.

Reply:
I agree -- I originally had a more descriptive error but scrapped it in favor of using the pandas one. Something that references mixed dtypes would be better